### PR TITLE
doc: cmake: Doc build improvements

### PR DESCRIPTION
**Pull request by carlescufi**
Minor doc build improvements

### CODE REVIEW

**Comment by carlescufi** [comment]
@dbkinder I spotted a difference in the generated .html with and without multicore builds:

File: .known-issues/doc/duplicate.conf [hunk 1]
```diff
@@ -4,3 +4,9 @@
 ^(?P<filename>([\-:\\/\w\.])+[/\\]doc[/\\]api[/\\]io_interfaces.rst):(?P<lineno>[0-9]+): WARNING: Duplicate declaration.
 #
 ^(?P<filename>([\-:\\/\w\.])+[/\\]doc[/\\]subsystems[/\\]sensor.rst):(?P<lineno>[0-9]+): WARNING: Duplicate declaration.
+#
+^(?P<filename>([\-:\\/\w\.])+[/\\]doc[/\\]api[/\\]io_interfaces.rst): WARNING: Duplicate declaration, also defined in \'subsystems/sensor\'\.
```
Comment (carlescufi) [inline .known-issues/doc/duplicate.conf]: @dbkinder I believe those are not real warnings, but can you please double-check?

Comment (carlescufi) [reply .known-issues/doc/duplicate.conf]: The actual warnings as they were shown by sphinx are:

```
/home/carles/src/zephyr/doc/api/io_interfaces.rst: WARNING: Duplicate declaration, also defined in 'subsystems/sensor'.
Declaration is 'sensor_value'.

/home/carles/src/zephyr/doc/api/io_interfaces.rst: WARNING: Duplicate declaration, also defined in 'subsystems/sensor'.
Name of declaration is 'sensor_value'.

/home/carles/src/zephyr/zephyr/doc/subsystems/sensor.rst: WARNING: Duplicate declaration, also defined in 'api/io_interfaces'.
Declaration is 'sensor_value'.
/home/carles/src/zephyr/zephyr/doc/subsystems/sensor.rst: WARNING: Duplicate declaration, also defined in 'api/io_interfaces'.
Name of declaration is 'sensor_value'.
```

Comment (dbkinder) [reply .known-issues/doc/duplicate.conf]: The "duplicate declaration" warning is a known "Sphinx/Breathe" issue as discussed in https://github.com/sphinx-doc/sphinx/issues/4820 and https://github.com/michaeljones/breathe/pull/377  
For now we're adding the pattern to ignore in our ".known_issues" system, sigh.

Comment (carlescufi) [reply .known-issues/doc/duplicate.conf]: Right, but it's strange that when enabling multicore builds these 2 new ones appeared. Might have something to do with the way sphinx builds.

Comment (carlescufi) [reply .known-issues/doc/duplicate.conf]: Thanks for the links btw

Comment (carlescufi) [reply .known-issues/doc/duplicate.conf]: @dbkinder funnily this PR is merged https://github.com/michaeljones/breathe/pull/377. Do we need to update our breath/sphinx versions?

**Comment by dbkinder** [comment]
Something isn't quite right in how inter-document linking is working in the multi-core builds.  For example in your diffs on the generated HTML, in ``api/io_interfaces.html``, the single core run makes a correct reference link to:

    <a class="reference internal" href="../subsystems/sensor.html#structsensor__value">

but the multi-core run makes an incorrect reference (``#structsensor__value`` is not a valid target in the **current** html file):

    <a class="reference internal" href="#structsensor__value">

Doesn't look like multi-core Breathe is working here :(

**Comment by carlescufi** [comment]
@dbkinder 
>Something isn't quite right in how inter-document linking is working in the multi-core builds.

Right, I suspected that but I did want someone to check. It's a shame but i'll remove the commit from this PR.

**Comment by carlescufi** [comment]
@dbkinder removed the commit. We can always try in a few months again, once we've updated to newer Sphinx and Breathe versions.

**Review by dbkinder — CHANGES_REQUESTED** [review]
In your diffs below, I see an incorrect href target link generated in the HTML for the multi-core run :(

**Comment by carlescufi** [comment]
>In your diffs below, I see an incorrect href target link generated in the HTML for the multi-core run :(

I removed the commit that enabled multicore builds.

**Comment by dbkinder** [comment]
@carlescufi A different approach for a faster "draft" doc build could be to have ``genrest.py`` behave differently when doing a "htmldraft" build.  Generate the doc/references/kconfig/index.rst with only the ``.. option::`` directives in it  (instead of in 3800 separate files) but no details, so the document build won't throw any errors/warnings because the uses of ``:option:`name` `` are undefined.

Then again, who's the target for this "htmldraft" build option?  If it's for someone editing a .rst file and wants to check that one file out to make sure it looks OK, there are alternatives to doing a whole website generation. Your previous optimization to not regenerate the separate .rst files for each option if the content didn't change works well for speeding up local doc generation for subsequent runs.  If we're trying to speed up the CI doc build part, we probably do need to generate the whole website.

**Comment by carlescufi** [comment]
>A different approach for a faster "draft" doc build could be to have genrest.py behave differently when doing a "htmldraft" build. Generate the doc/references/kconfig/index.rst with only the .. option:: directives in it (instead of in 3800 separate files) but no details, so the document build won't throw any errors/warnings because the uses of :option:`name` are undefined.

@dbkinder 
I agree that this might have been a bit rushed from my side. I have removed this change from this PR and instead I only now have minor fixes.
I will attempt a better approach to a faster draft build based on `genrest.py` at a later time, but we can merge this PR in the meantime.

**Review by dbkinder — APPROVED** [review]
Looks OK to me :)
(But commit message needs attention)

**Pull request merged**